CMakeLists.txt

```diff
@@ -686,6 +704,9 @@ if (MFEM_USE_MPI)
     target_link_libraries(mfem PUBLIC ${MPI_CXX_LINK_FLAGS})
   endif()
 endif()
+if (MFEM_USE_ENZYME)
```

Comment:
`ENZYME` should be removed from L599 since the `target_link_libraries()` call is here.  Note adding `ENZYME` to the list of `MFEM_TPLS` doesn't do anything because the package name is `Enzyme`.  Also, FYI, on my install, changing `ENZYME` to `Enzyme` on L599 didn't work, since the `Enzyme_LIBRARIES` variable had a lot of weird things in it.